openmindat/localities.py

```diff
@@ -331,46 +331,144 @@ def updated_at(self, DATE_STR):
 
         return self
     
-    def saveto(self, OUTDIR=''):
+    def saveto(self, OUTDIR = '', FILE_NAME = ''):
         '''
-            Executes the query to retrieve the geomaterials with keywords and saves the results to a specified directory.
+            Executes the query to retrieve the localities with keywords and saves the results to a specified directory.
 
             Args:
-                OUTDIR (str): The directory path where the retrieved geomaterials will be saved. If not provided, the current directory will be used.
+                OUTDIR (str): The directory path where the retrieved localities will be saved. If not provided, the current directory will be used.
+                FILE_NAME (str): An optional file name, if no input is given it uses the end point as a name
 
             Returns:
                 None
 
             Example:
                 >>> lr = LocalitiesRetriever()
-                >>> lr.saveto("/path/to/directory")
+                >>> lr.saveto("/path/to/directory", "france")
         '''
 
         print("Retrieving localities. This may take a while... ")
 
         params = self._params
         end_point = 'localities'
         outdir = OUTDIR
+        file_name = FILE_NAME
 
         ma = mindat_api.MindatApi()
-        ma.get_mindat_list(params, end_point, outdir)
+        ma.get_mindat_list(params, end_point, outdir, file_name)
 
         # reset the query parameters in case the user wants to make another query
         self._init_params()
     
-    def save(self):
+    def save(self, FILE_NAME = ''):
         '''
-            Executes the query to retrieve the list of geomaterials and saves the results to the current directory.
+            Executes the query to retrieve the list of localities and saves the results to the current directory.
 
+            Args:
+                FILE_NAME (str): An optional file name, if no input is given it uses the end point as a name
+            
             Returns:
                 None
 
             Example:
                 >>> lr = LocalitiesRetriever()
                 >>> lr.save()
         '''
-        self.saveto()
+        file_name = FILE_NAME
+        
+        self.saveto('', file_name)
+        
+        
+class LocalitiesIdRetriever:
+    """
+    This module provides the LocalitiesIdRetriever class for returning localities by id
+
+    Usage:
+        >>> lir = LocalitiesIdRetriever()
+        >>> lir.id(5)
+
+    Attributes:
+        _id (int): An int to store id parameter.
```

Comment:
The initial underscore before the parameters of functions indicates that the object is a private target. However, this naming cannot prevent outside functions from calling the parameter. It is just a programming style for readability. So, please avoid using the initial underscore for non-private objects.